holopy/core/io/io.py

```diff
@@ -413,31 +413,57 @@ def load_average(filepath, refimg=None, spacing=None, medium_index=None, illum_w
     if channel is None and refimg is not None and illumination in refimg.dims:
         channel = [i for i, col in enumerate(['red','green','blue']) if col in refimg[illumination].values]
 
-    accumulator = clean_concat([load_image(image, spacing, channel=channel) for image in filepath],'images')
-
     if np.isscalar(spacing):
         spacing = np.repeat(spacing, 2)
 
+    # calculate the average
+    accumulator = Accumulator()
+    for path in filepath:
+        accumulator.push(load_image(path, spacing, channel=channel))
+    mean_image = accumulator.mean()
+
+    # calculate average noise from image
+    if noise_sd is None and len(filepath) > 1:
+        noise_sd = ensure_array(accumulator.std())
+
     # crop according to refimg dimensions
     if refimg is not None:
         def extent(i):
             name = ['x','y'][i]
             return np.around(refimg[name].values/spacing[i]).astype('int')
-        accumulator = accumulator.isel(x=extent(0), y=extent(1))
-        accumulator['x'] = refimg.x
-        accumulator['y'] = refimg.y
-
-    # calculate the average
-    mean = accumulator.mean('images')
-
-    # calculate average noise from image
-    if noise_sd is None and len(filepath) > 1:
-        noise_sd = ensure_array((accumulator.std('images')/mean).mean(('x','y','z')))
-    accumulator = mean
+        mean_image = mean_image.isel(x=extent(0), y=extent(1))
+        mean_image['x'] = refimg.x
+        mean_image['y'] = refimg.y
 
     # copy metadata from refimg
     if refimg is not None:
-        accumulator = copy_metadata(refimg, accumulator, do_coords=False)
+        mean_image = copy_metadata(refimg, mean_image, do_coords=False)
 
     # overwrite metadata from refimg with provided values
-    return update_metadata(accumulator, medium_index, illum_wavelen, illum_polarization, normals, noise_sd)
+    return update_metadata(mean_image, medium_index, illum_wavelen, illum_polarization, normals, noise_sd)
+
+class Accumulator:
+    def __init__(self):
+        self.n = 0
+        self.m = None
+        self.s = None
```

Comment:
I think this accumulator class is great, but can I just ask for longer variable names, e.g. ```self._running_mean``` instead of ```self.m```? Also, ```self.n, self.m, self.s``` are "private", right?

Reply:
Fixed by commits 42da7e3 and c1c08d4